libethereum/ChainParams.cpp

```diff
@@ -116,8 +116,9 @@ ChainParams ChainParams::loadGenesis(string const& _json, h256 const& _stateRoot
 SealEngineFace* ChainParams::createSealEngine()
 {
 	SealEngineFace* ret = SealEngineRegistrar::create(sealEngineName);
-	assert(ret && "Seal engine not found");
-	ret->setChainParams(*this);
+	assert((ret != nullptr) && "Seal engine not found");
+	if (ret != nullptr)
```

Comment:
Why?   That is pretty poor advice, IMHO.
And where do you see that recommendation?